radicle-cli/src/commands/patch/checkout.rs

```diff
@@ -0,0 +1,43 @@
+use crate::terminal as term;
+use radicle::cob::patch::{self, PatchId};
+use radicle::git;
+use radicle::profile::Profile;
+use radicle::storage::git::Repository;
+
+pub fn run(
+    storage: &Repository,
+    profile: &Profile,
+    git_workdir: &git::raw::Repository,
+    patch_id: &PatchId,
+) -> anyhow::Result<()> {
+    let patchs = patch::Patches::open(profile.public_key, storage)?;
+    let Some(patch) = patchs.get(patch_id)? else {
+        anyhow::bail!("Patch `{patch_id}` not found");
+    };
+
+    let spinner = term::spinner("Performing patch checkout...");
+
+    // Getting the patch obj!
+    let patch_haed = patch.head();
+    let obj = git_workdir.revparse_single(&patch_haed.to_string())?;
+
+    let branch_name = format!("patch/{}", term::format::cob(patch_id));
+    // checkout the patch in a new branch!
+    let commit = if let Some(commit) = obj.as_commit() {
+        commit
+    } else {
+        anyhow::bail!("Unable to convert object to a commit.");
+    };
+    git_workdir.branch(&branch_name, commit, false)?;
+    // and then point the current `HEAD` inside the new branch.
+    git_workdir.set_head(format!("refs/heads/{branch_name}").as_str())?;
```

Comment:
I think it would be nicer if the functionality allowed me to specify the branch. The patch id could be a default. If we were to do that, I would make sure to strip `refs/heads` from the input if provided.

Reply:
>I would make sure to strip refs/heads from the input if provided.

Good catch, thanks!

>I think it would be nicer if the functionality allowed me to specify the branch. The patch id could be a default.

Yeah as I suggest in the PR description, but I did not know if there was some previous discussion about it so I propose the naive one